tools/build-toolchain.sh

```diff
@@ -38,35 +78,96 @@ command_exists () {
+  if   command_exists wget ; then wget --no-check-certificate -c  "$1"
   elif command_exists curl ; then curl -LO "$1"
   else
-    echo "Install `wget` or `curl` to download toolchain sources" 1>&2
+    echo "Install 'wget' or 'curl' to download toolchain sources" 1>&2
     return 1
   fi
 }
 
-# Dependency source: Download stage
+echo "Stage: Download and extract dependencies"
 test -f "binutils-$BINUTILS_V.tar.gz" || download "https://ftp.gnu.org/gnu/binutils/binutils-$BINUTILS_V.tar.gz"
+test -d "binutils-$BINUTILS_V"        || tar -xzf "binutils-$BINUTILS_V.tar.gz"
+
 test -f "gcc-$GCC_V.tar.gz"           || download "https://ftp.gnu.org/gnu/gcc/gcc-$GCC_V/gcc-$GCC_V.tar.gz"
+test -d "gcc-$GCC_V"                  || tar -xzf "gcc-$GCC_V.tar.gz" --checkpoint=.100 # TODO: there must be a better way of showing progress (given such a large file)!
+
 test -f "newlib-$NEWLIB_V.tar.gz"     || download "https://sourceware.org/pub/newlib/newlib-$NEWLIB_V.tar.gz"
+test -d "newlib-$NEWLIB_V"            || tar -xzf "newlib-$NEWLIB_V.tar.gz"
+
+# Optional dependency handling
+# Copies the FP libs into GCC sources so they are compiled as part of it
+if [ "$GMP_V" != "" ]; then
+  test -f "gmp-$GMP_V.tar.xz"         || download "https://ftp.gnu.org/gnu/gmp/gmp-$GMP_V.tar.xz"
+  test -d "gmp-$GMP_V"                || tar -xf "gmp-$GMP_V.tar.xz" # note no .gz download file currently available
+  cd "gcc-$GCC_V"
+  ln -sf ../"gmp-$GMP_V" "gmp"
+  cd ..
+fi
+if [ "$MPC_V" != "" ]; then
+  test -f "mpc-$MPC_V.tar.gz"         || download "https://ftp.gnu.org/gnu/mpc/mpc-$MPC_V.tar.gz"
+  test -d "mpc-$MPC_V"                || tar -xzf "mpc-$MPC_V.tar.gz"
+  cd "gcc-$GCC_V"
+  ln -sf ../"mpc-$MPC_V" "mpc"
+  cd ..
+fi
+if [ "$MPFR_V" != "" ]; then
+  test -f "mpfr-$MPFR_V.tar.gz"       || download "https://ftp.gnu.org/gnu/mpfr/mpfr-$MPFR_V.tar.gz"
+  test -d "mpfr-$MPFR_V"              || tar -xzf "mpfr-$MPFR_V.tar.gz"
+  cd "gcc-$GCC_V"
+  ln -sf ../"mpfr-$MPFR_V" "mpfr"
+  cd ..
+fi
+# Certain platforms might require Makefile cross compiling
+if [ "$MAKE_V" != "" ]; then
+  test -f "make-$MAKE_V.tar.gz"       || download "https://ftp.gnu.org/gnu/make/make-$MAKE_V.tar.gz"
+  test -d "make-$MAKE_V"              || tar -xzf "make-$MAKE_V.tar.gz"
+fi
 
-# Dependency source: Extract stage
-test -d "binutils-$BINUTILS_V" || tar -xzf "binutils-$BINUTILS_V.tar.gz"
-test -d "gcc-$GCC_V"           || tar -xzf "gcc-$GCC_V.tar.gz"
-test -d "newlib-$NEWLIB_V"     || tar -xzf "newlib-$NEWLIB_V.tar.gz"
+if [ "$BUILD" != "$HOST" ]; then
+  echo "Stage: Patch step"
+  
+  if [[ "$GCC_V" = "11.2.0" || "$GCC_V" = "11.1.0" ]]; then
+    # GCC 11.x fails on canadian cross
+    # see: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=100017
+    # see: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=80196
+    echo "Apply patch for GCC using SED:"
+    sed -z 's/RAW_CXX_FOR_TARGET="$CXX_FOR_TARGET"/RAW_CXX_FOR_TARGET="$CXX_FOR_TARGET -nostdinc++"/' ./"gcc-$GCC_V"/configure
```

Comment:
In general, I think it would be easier to track these as `.patch` files, committed in the source tree, and applied with `patch`.